### PR TITLE
Update Toolchain/Android-Target to v24

### PR DESCRIPTION
**Pull request by AndyScherzinger**
bump everything to 24 so the app project can then move to v24 to support Android N.

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
cc: @tobiasKaminsky and @nextcloud/android for review.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky is it okay for you if I merge this, create a new version 1.0.5 to use in my app branch to work on the general v24 version bump?

**Comment by tobiasKaminsky** [comment]
Let's do in parallel with app 1.4.0.
Maybe there will be already android N support ;)

File: test_client/tests/res/values/strings.xml [hunk 1]
```diff
@@ -25,6 +25,6 @@
  
 <resources>
 
-    <string name="app_name">Oc_framework-testTest</string>
+    <string name="app_name">Nc_framework-testTest</string>
```
Comment (tobiasKaminsky) [inline test_client/tests/res/values/strings.xml]: nice catch ;)

**Comment by tobiasKaminsky** [comment]
code wise :+1: 
We have to test it extensively.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      - echo no | android create avd --force -n test -t $ANDROID_TARGET --abi $ANDROID_ABI -c 20M
+      - emulator -avd test -no-window &
+      - ./wait_for_emulator.sh
+      # build app
+      - ./gradlew clean build
+
+    environment:
+      - ANDROID_TARGET=android-24
+      - ANDROID_ABI=armeabi-v7a
+      - LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/:/opt/android-sdk-linux/tools/lib64/gles_mesa/
```
Comment (tobiasKaminsky) [inline .drone.yml]: Is this whole drone.yml up to date?

Comment (AndyScherzinger) [inline .drone.yml]: Will have to check, but I thought so

Comment (AndyScherzinger) [inline .drone.yml]: Yes, this one is fine :)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky anything else to do here in your opinion?

**Comment by AndyScherzinger** [comment]
merging... :+1: 

The thing missing in the build file is 
```
- ant clean
- ant debug
- cd test_client/tests
- ant acceptance-test
- cd ../..
```
since we do not have a server yet to run integration tests on...

**Pull request merged**